compiler/luci/pass/src/CircleOptimizer.cpp

```diff
@@ -279,318 +284,165 @@ void CircleOptimizer::optimize(loco::Graph *g) const
+    &createPassInstance<luci::SubstitutePackToReshapePass>;
+  option_to_pass[Options::Algorithm::SubstitutePadV2ToPad] =
+    &createPassInstance<luci::SubstitutePadV2ToPadPass>;
+  option_to_pass[Options::Algorithm::SubstituteSplitVToSplit] =
+    &createPassInstance<luci::SubstituteSplitVToSplitPass>;
+  option_to_pass[Options::Algorithm::SubstituteSqueezeToReshape] =
+    &createPassInstance<luci::SubstituteSqueezeToReshapePass>;
+  option_to_pass[Options::Algorithm::SubstituteStridedSliceToReshape] =
+    &createPassInstance<luci::SubstituteStridedSliceToReshapePass>;
+  option_to_pass[Options::Algorithm::SubstituteTransposeToReshape] =
+    &createPassInstance<luci::SubstituteTransposeToReshapePass>;
+  option_to_pass[Options::Algorithm::TransformMinMaxToRelu6Pass] =
+    &createPassInstance<luci::TransformMinMaxToRelu6Pass>;
+  option_to_pass[Options::Algorithm::TransformMinReluToRelu6Pass] =
+    &createPassInstance<luci::TransformMinReluToRelu6Pass>;
+  option_to_pass[Options::Algorithm::TransformSqrtDivToRsqrtMul] =
+    &createPassInstance<luci::TransformSqrtDivToRsqrtMulPass>;
+  option_to_pass[Options::Algorithm::DecomposeHardSwishPass] =
+    &createPassInstance<luci::DecomposeHardSwishPass>;
+  option_to_pass[Options::Algorithm::DecomposeSoftmaxPass] =
+    &createPassInstance<luci::DecomposeSoftmaxPass>;
+  option_to_pass[Options::Algorithm::UnrollUnidirSeqLSTM] =
+    &createPassInstance<luci::UnrollUnidirectionalSequenceLSTMPass>;
   // NOTE Experimental options; these will be removed someday
   //      Add experimental options here
-  if (_options->query(Options::Algorithm::XpSepActFromTransposeConv))
-  {
-    phase.emplace_back(std::make_unique<luci::XpSepActFromTransposeConvPass>());
-  }
-
+  option_to_pass[Options::Algorithm::XpSepActFromTransposeConv] =
+    &createPassInstance<luci::XpSepActFromTransposeConvPass>;
   // Forward Reshape/Transpose is done after
   // 1. SubstituteXXXToReshape
   // 2. RemoveRedundantReshape/Transpose
   // See https://github.com/Samsung/ONE/pull/10596 for more details
-  if (_options->query(Options::Algorithm::ForwardReshapeToUnaryOp))
+  option_to_pass[Options::Algorithm::ForwardReshapeToUnaryOp] =
+    &createPassInstance<luci::ForwardReshapeToUnaryOpPass>;
+  option_to_pass[Options::Algorithm::ForwardTransposeOp] =
+    &createPassInstance<luci::ForwardTransposeOpPass>;
+
+  for (auto const &m : option_to_pass)
   {
-    phase.emplace_back(std::make_unique<luci::ForwardReshapeToUnaryOpPass>());
+    if (_options->query(m.first))
+    {
+      phase.emplace_back(m.second());
+    }
   }
-  if (_options->query(Options::Algorithm::ForwardTransposeOp))
+
+  if (_options->query(Options::Algorithm::RemoveUnnecessaryReshape))
   {
-    phase.emplace_back(std::make_unique<luci::ForwardTransposeOpPass>());
+    phase.emplace_back(std::make_unique<luci::RemoveUnnecessaryReshapePass>());
+    phase.emplace_back(std::make_unique<luci::RemoveUnnecessaryReshapeNetPass>());
```

Comment:
maybe put this before `// NOTE Experimental options; these will be removed someday` line to match order?

Reply:
@seanshpark Right. This is because `RemoveUnnecessaryReshape` option needs two instantiation of the passes:  `RemoveUnnecessaryReshapePass`, `RemoveUnnecessaryReshapeNetPass`. But introduced map only takes one pass per one option. 

We can make it possible but should be a bit more complicated and the case is only one as of now. Maybe taking some rain check on this seems good..

Reply:
After some offline talk, I realized some passes have dependent on previous passes. I'll make a map only for independent passes.